core/src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -19,94 +19,163 @@
     private final String queryText;
+    /** Boost to apply to resulting Lucene query. */
+    private float boost = 1.0f;
+    /**
+     * Fields to query against. If left empty will query default field,
+     * currently _ALL. Uses a TreeMap to hold the fields so boolean clauses are
+     * always sorted in same order for generated Lucene query for easier
+     * testing.
+     * 
+     * Can be changed back to HashMap once https://issues.apache.org/jira/browse/LUCENE-6305 is fixed.
+     */
+    private final Map<String, Float> fieldsAndWeights = new TreeMap<>();
+    /** If specified, analyzer to use to parse the query text, using registered default if left empty. */
+    private String analyzer;
+    /** Name of the query. */
     private String queryName;
+    /** Default operator to use for linking boolean clauses. */
+    private Operator defaultOperator;
+    /** If result is a boolean query, minimumShouldMatch parameter to apply. Ignored otherwise. */
     private String minimumShouldMatch;
+    /** Any search flags to be used, ALL by default. */
     private int flags = -1;
-    private Boolean lowercaseExpandedTerms;
-    private Boolean lenient;
-    private Boolean analyzeWildcard;
-    private Locale locale;
+    /** Further search settings. */
+    private Settings settings = new Settings();
+
     static final SimpleQueryStringBuilder PROTOTYPE = new SimpleQueryStringBuilder(null);
 
-    /**
-     * Operators for the default_operator
-     */
+    /** Operators available for linking boolean clauses. */
     public static enum Operator {
         AND,
-        OR
+        OR;
+
+        public static Operator parseFromInt(int ordinal) {
+            switch (ordinal) {
+            case 0:
+                return AND;
+            case 1:
+                return OR;
+            default:
+                throw new IllegalStateException("Cannot parse Operator from ordinal " + ordinal);
+            }
+                
+        }
     }
 
-    /**
-     * Construct a new simple query with the given text
-     */
-    public SimpleQueryStringBuilder(String text) {
-        this.queryText = text;
+    /** Construct a new simple query with this query string. */
+    public SimpleQueryStringBuilder(String queryText) {
+        this.queryText = queryText;
     }
 
-    /**
-     * Add a field to run the query against
-     */
+    /** Returns the text to parse the query from. */
+    public String text() {
+        return this.queryText;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public SimpleQueryStringBuilder boost(float boost) {
+        this.boost = boost;
+        return this;
+    }
+    
+    /** Returns the boost to apply to resulting Lucene query.*/
+    public float boost() {
+        return this.boost;
+    }
+
+    /** Add a field to run the query against. */
     public SimpleQueryStringBuilder field(String field) {
-        this.fields.put(field, null);
+        if (! Strings.isEmpty(field)) {
```

Comment:
actuallt are we sure it's good to just ignore it if field is null? I would rather leave as it was (it throws NPE) and throw IAE ourselves, not much difference though